runtime/autoload/provider/node.vim

```diff
@@ -83,6 +83,17 @@ function! provider#node#Detect() abort
     let pnpm_opts.job_id = jobstart('pnpm --loglevel silent root -g', pnpm_opts)
   endif
 
+  let bun_opts = {}
+  if executable('bun')
+    let bun_opts = deepcopy(s:NodeHandler)
+    let bun_opts.entry_point = '/node_modules/neovim/bin/cli.js'
+    let bun_config_dir = '/.bun'
+    let bun_default_path = $HOME . bun_config_dir . '/install' . '/global' . bun_opts.entry_point
```

Comment:
So it resolves to the bun's global installation path: `$HOME/.bun/install/global/node_modules/`

But it also needs to respect `$BUN_INSTALL`, or any other custom globalDir as per [bun config](https://bun.sh/docs/runtime/bunfig#install-globaldir). Is there any way to query `burn` (as we ask npm, or yarn, etc.) to get this path?

Reply:
Instead we can query `bun pm ls -g` to see if neovim is installed globally via bun.

Reply:
I went through the documentation (more carefully this time), read some help pages, and looked at some issues, but I did not find any command that gives the output we want. I also sent a message on their discord, so if there is a way to do it, we'll probably find out.

`bun pm ls -g` does show the path, but the output is not clear. It would be necessary to apply some filtering/parsing to get the right information. Even though that works, any change to the output would most likely just break the thing. This gives us two 'unstable' options (hardcoded and filtered output). We could go with one of them or just not merge this at all, at least for the moment.

I should have gone through this before making the PR. My bad.

Reply:
No worries! That's what reviews and PRs are for. Thanks for writing a great enhancement 👍️!

Reply:
@wookayin I found a way. It was actually very simple and I was just being kinda dumb. `bun pm bin -g` gives the path to the globally installed binaries dir. In that directory, there is a symlink to `cli.js` located in the neovim module. It seems that bun creates a symlink in the binaries dir for the path specified in the bin key in the `package.json`.

I'll turn this into a draft PR and make the changes. The path will end up like this:

`[bin_path]/neovim-node-host` where `bin_path` is the output of `bun pm bin -g`.

Sounds good?

Reply:
It works, but is kind of a hack because it points to `bin/neovim-node-host` not `node_modules/neovim`. In other paths not via bun, `entry_point` is set to be `node_modules/neovim/bin/cli.js`. I think a situation where `bin/neovim-node-host` exists but `node_modules/neovim` are broken, or *vice versa*, is an exceptional case that we shouldn't worry too much about, but we may want to be consistent. I have no strong preference yet, but will think about what's the best way to deal with this.

BTW the documentation around `g:node_host_prog` also needs to be updated.

Reply:
@justinmk any advice?

Reply:
Seems fine. As long as something (`:checkhealth` at least) checks that it actually works, and reports the discovered paths.